File: readability/readability.c
//Libraries
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

//Declare functions
int count_letters(string t);
int count_words(string t);
int count_sentences(string t);

int main(void)
{
    //Get text input from user
    string text = get_string("Text: ");

    //Assign counts to variables
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);

    //Calculate index
    float l = letters / words * 100;
    float s = sentences / words * 100;
    float index = 0.0588 * l - 0.296 * s -  15.8;

    //If index is 16 or above, print out "Grade 16+"
    if (index > 15)
    {
        printf("Grade 16+\n");
    }
    //If index is less than 1, print out "Before Grade 1"
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    //Otherwise, print out index as nearest whole number
    else
    {
        printf("Grade %1.f\n", index);
    }
}

//Count alphabetical chars
int count_letters(string t)
{
    int count = 0;
    for (int i = 0, l = strlen(t); i < l; i++)
    {
        if (isalpha(t[i]))
        {
            count++;
        }
    }
    return count;
}

//Count spaces and infer word count per problem set instructions
int count_words(string t)
{
    int count = 0;
    for (int i = 0, l = strlen(t); i < l; i++)
    {
        if (isspace(t[i]))
        {
            count++;
        }
    }
    return count + 1;
}

//Count ".", "!", and "?" and infer sentence count per problem set instructions
int count_sentences(string t)
{
    int count = 0;
    for (int i = 0, l = strlen(t); i < l; i++)
    {
        if (t[i] == '.' || t[i] == '!' || t[i] == '?')
        {
            count++;
        }
    }
    return count;
}

File: pset6/readability/readability.py
from cs50 import get_string
import re

# Get text input from user
text = get_string("Text: ")

# Get length of list words in text
words = len(re.split('\s', text))

# Get length of list of sentences in text
# Subtract 1 due to split giving extra string at end
sentences = len(re.split('[.!?]', text)) - 1

# Get rid of spaces and punctuation in text
letters = text.replace(" ", "")
letters = letters.replace(".", "")
letters = letters.replace(",", "")
letters = letters.replace("!", "")
letters = letters.replace("'", "")
letters = letters.replace("\"", "")

# Get length of list of only characters in text
letters = len(letters.replace("?", ""))

# DEBUG
# print(letters, words, sentences)

# Apply Coleman-Liau index
l = letters / words * 100
s = sentences / words * 100
index = 0.0588 * l - 0.296 * s - 15.8

# DEBUG
# print(l, s, index)

# Output grade level
if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print("Grade", int(round(index)))